client/src/main/scala/com/raquo/app/todomvc/TodoMvcApp.scala

```diff
@@ -46,14 +50,21 @@ object TodoMvcApp {
 
   // Var-s are reactive state variables suitable for both local state and redux-like global stores.
   // Laminar uses my library Airstream as its reactive layer https://github.com/raquo/Airstream
+  private val ls = window.localStorage.getItem("todos-laminar")
+  private val initItems: List[TodoItem] = if(ls == null)
+    List.empty[TodoItem]
+  else
+    Json.decode(ls.getBytes("UTF8")).to[List[TodoItem]].value
 
-  private val itemsVar = Var(List[TodoItem]())
+  private val itemsVar = Var(initItems)
   private val filterVar = Var[Filter](ShowAll)
   private val toggleAllVar = Var[Boolean](false)
   private var lastId = 1 // just for auto-incrementing IDs
 
   private def updateToggleAll() = toggleAllVar.update(_ => itemsVar.now().nonEmpty && itemsVar.now().forall(_.completed))
 
+  private val localStorageObserver = Observer[List[TodoItem]](onNext = items => window.localStorage.setItem("todos-laminar", Json.encode(items).toUtf8String))
```

Comment:
* `localStorageWriter` would be a better name (similar to Var.writer), "observer" in the name implies that it observes local storage instead
* Please move `ls` and `initItems` down here to keep localStorage logic in one place
* Please extract "todos-laminar" into `private val localStorageKey`
* Please use `dom.window` instead of `window`. Laminar's recommended style is to never import anything from under `dom`, so that it's always clear when we're talking to native JS DOM APIs.